### PR TITLE
[JENKINS-67674] Update bundled dependencies after the advisory

### PR DESCRIPTION
**Pull request by Wadeck**
Motivated by https://www.jenkins.io/security/advisory/2022-01-12/

### Proposed changelog entries

* Update the bundled Mailer Plugin from 1.32.1 to 408.vd726a_1130320 based on [security advisory](https://www.jenkins.io/security/advisory/2022-01-12/).
* Update the bundled Matrix Project Plugin from 1.18 to 1.20 based on [security advisory](https://www.jenkins.io/security/advisory/2022-01-12/).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by MarkEWaite** [comment]
This would supersede #6177 and #6178 that are both marked as "ready for merge", waiting for the usual 24 hour timeout after approvals have been received.  Is there a compelling reason why we should combine those two pull requests into this pull request?

**Comment by Wadeck** [comment]
@MarkEWaite different pom.xml no? this one is for war/pom.xml, the two others are for test/pom.xml

TOO_MANY_POM_EXCEPTION? :D

**Review by MarkEWaite — APPROVED** [review]

File: war/pom.xml
Comment (basil) [inline war/pom.xml:291]: Upgrading Matrix Project to 1.20 also requires upgrading JUnit to 1.47 or higher, but JUnit 1.29 is bundled a few lines below. I suspect this will cause `LoadDetachedPluginsTest` to fail.

Comment (Wadeck) [reply war/pom.xml:291]: Thanks for the insights, if the build failed I will have a look next week.

Comment (basil) [reply war/pom.xml:291]: As predicted: https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6180/2/testReport/junit/jenkins.install/LoadDetachedPluginsTest/Linux_jdk11___Linux_Build___Test___upgradeFromJenkins1/

**Comment by timja** [comment]
@daniel-beck 

```
Caused by: java.io.IOException: Failed to load: JUnit Plugin (junit 1.47)
 - Plugin is missing: plugin-util-api (1.5.0)
 - Plugin is missing: bootstrap4-api (4.5.3-1)
 - Plugin is missing: checks-api (1.1.1)
 - Plugin is missing: display-url-api (2.3.4)
 - Plugin is missing: jackson2-api (2.12.0)
 - Plugin is missing: echarts-api (4.9.0-2)
```

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
